lms/resources/_js_config/__init__.py

```diff
@@ -87,6 +87,12 @@ def add_document_url(self, document_url):
                     _query={"document_url": document_url},
                 ),
             }
+        elif document_url.startswith("canvas://"):
+            course_id = self._request.params["custom_canvas_course_id"]
+            file_id = self._request.params["file_id"]
+            resource_link_id = self._request.params["resource_link_id"]
```

Comment:
The next PR changes the via url to take https://github.com/hypothesis/lms/pull/3186 document URL directly.

Reply:
Ok, so calling `JSConfig.add_document_url(document_url)` with a `canvas://file/` URL does the same thing as calling `JSConfig.add_canvas_file_id(course_id, resource_link_id, file_id)` would do. And I can see that this fits with the already-existing special-casing of `blackboard://` URLs in this method :+1:

It's weird that `add_document_url(document_url)` actually ignores the given `document_url` (other than doing `document_url.startswith("canvas://file/")`) and uses `request.params` instead. But I see that https://github.com/hypothesis/lms/pull/3186/ changes it to use `document_url` instead of `request.params`